src/test/java/com/synopsys/integration/stepworkflow/StepWorkflowTest.java

```diff
@@ -17,26 +22,62 @@
+    private SubStep<Integer, Object> subStepFunctionTwo;
+    private StepWorkflow.FlowController<Object, Integer> flowControllerOne;
+    private StepWorkflow.Builder<Integer> successfulBuilderOfSupplierOne;
+    private StepWorkflow.Builder<Integer> builderThenSubStepSupplierTwo;
 
-    private final SubStep<Object, Object> successfulExecutorOne = SubStep.ofExecutor(this::successfulExecutor);
-    private final SubStep<Integer, Object> successfulFunctionOne = SubStep.ofFunction(this::successfulFunctionOne);
-    private final SubStep<Integer, Object> successfulFunctionTwo = SubStep.ofFunction(this::successfulFunctionTwo);
+    private StepWorkflow.Builder<Integer> unsuccessfulBuilderOfSupplierOne;
+    private SubStep<Object, Integer> unsuccessfulSubStepSupplierOne;
 
-    private final StepWorkflow.FlowController<Object, Integer> successfulFlowControllerOfSupplier = new StepWorkflow.FlowController<>(successfulSupplierOne);
+    private static Stream<Boolean> runConditionalWorkflowObjectToTest() {
+        return Stream.of(true, false);
+    }
 
-    private final StepWorkflow.Builder<Integer> successfulBuilderOfSupplierOne = StepWorkflow.first(successfulSupplierOne);
-    private final StepWorkflow.Builder<Integer> unsuccessfulBuilderOfSupplierOne = StepWorkflow.first(unsuccessfulSupplierOne);
+    @BeforeEach
+    public void setup() {
+        subStepSupplierOne = SubStep.ofSupplier(this::successfulDataSupplierOne);
+        subStepSupplierTwo = SubStep.ofSupplier(this::successfulDataSupplierTwo);
+        subStepSupplierThree = SubStep.ofSupplier(this::successfulDataSupplierThree);
+        subStepExecutorOne = SubStep.ofExecutor(this::successfulExecutor);
+        subStepFunctionOne = SubStep.ofFunction(this::successfulFunctionOne);
+        subStepFunctionTwo = SubStep.ofFunction(this::successfulFunctionTwo);
+
+        flowControllerOne = new StepWorkflow.FlowController<>(subStepSupplierOne);
+        assertEquals(subStepSupplierOne, flowControllerOne.step);
+        assertNull(flowControllerOne.next);
+        assertNull(flowControllerOne.getResponse());
+
+        successfulBuilderOfSupplierOne = StepWorkflow.first(subStepSupplierOne);
+        assertEquals(subStepSupplierOne, successfulBuilderOfSupplierOne.start.step);
+        assertEquals(subStepSupplierOne, successfulBuilderOfSupplierOne.end.step);
+        assertNull(successfulBuilderOfSupplierOne.start.next);
+        assertNull(successfulBuilderOfSupplierOne.end.next);
+
+        builderThenSubStepSupplierTwo = successfulBuilderOfSupplierOne.then(subStepSupplierTwo);
+        assertEquals(subStepSupplierOne, builderThenSubStepSupplierTwo.start.step);
+        assertEquals(subStepSupplierTwo, builderThenSubStepSupplierTwo.end.step);
+        assertEquals(subStepSupplierTwo, builderThenSubStepSupplierTwo.start.next.step);
+        assertNull(builderThenSubStepSupplierTwo.end.next);
+
+        unsuccessfulSubStepSupplierOne = SubStep.ofSupplier(this::unsuccessfulDataSupplier);
```

Comment:
The unsuccessful substep should also be initialized as a final variable instead of remade before each test.